changes/3995-healarconr.md

```diff
@@ -0,0 +1 @@
+Fix mypy error in models_recursive.py example.
```

Comment:
It's not needed to add a change file for a small fix in the documentation. please remove it.